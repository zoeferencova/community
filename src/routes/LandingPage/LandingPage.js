import React, { Component } from "react";
import { Link } from "react-router-dom";

import styles from "./LandingPage.module.css";

export default class LandingPage extends Component {
    render() {
        return (   
            <main>
                <header className={styles.header}>
                    <h1>CommUnity</h1>
                    <h2>Coming together in a time of need</h2>
                    <p>Right now, more than ever, we need to unite our communities and help those in need. CommUnity is a free platform that enables the sharing of services and resources to ensure no one is left stranded during the Coronavirus (COVID-19) outbreak.</p>
                    <Link to="/register"><button>Start Now</button></Link>
                </header>
                <section className={styles.section}>
                    <h3>How does it work?</h3>
                    <p>[<em>placeholder for screenshot of main feed page</em>]</p>
                    <p>CommUnity lets users post offers or requests for help and see other active postings in their neighborhood.</p>
                </section>
                <section className={styles.section}>
                    <h3>What can our helpers help with?</h3>
                    <p>[<em>placeholder for graphic related to various services offered</em>]</p>
                    <p>Our helpers can assist with picking up supplies, running errands, or even just a friendly phone call or chat.</p>
                </section>
                <section className={styles.section}>
                    <h3>Safety and security are important to us</h3>
                    <p>[<em>placeholder for screenshot of chat/map</em>]</p>
                    <p>Your exact address will never be shared with other users, just your first name and neighborhood. Once matched with a helper or person requesting help, you will enter a private chat with the user where you can securely discuss details of the request.</p>
                </section>
                <footer className={styles.footer}>Footer</footer>
            </main>
        )
    }
}