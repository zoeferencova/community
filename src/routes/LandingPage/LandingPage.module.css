

.header, .section {
    display: flex;
    align-items: center;
    text-align: left;
}
  
.header {
    height: 100vh;
    background-color: #F5F1EA;
    position: absolute;
    top: 0;
}

.header h1 {
    padding-top: 100px;
}

.banner {
    height: 100vh;
    position: absolute;
    top: 0;
    right: 0px;
    z-index: 0;
    opacity: 0.9;
}

.header .left {
    z-index: 10;
    background-color: #F5F1EA;
}

.first {
    margin-top: 90vh;
}

.section {
    padding: 80px 0px;
}

.icons {
    flex-direction: column;
}
  
.right, .left {
    width: 50%;
    height: 100%;
    display: flex;
    justify-content: center;
    flex-direction: column;
}

.heading, .subheading, .buttonSection {
    width: 70%;
    margin: 0px auto 40px auto;
    z-index: 30;
}

.heading {
    font-size: 48px;
    padding-bottom: 0px;
}

.subheading {
    font-size: 20px;
    line-height: 25px;
}

.demoButton, .signupButton {
    border: 1px solid black;
    border-radius: 3px;
    padding: 12px 15px !important;
    font-family: inherit;
    font-size: 16px;
    letter-spacing: 0.7px;
    margin-right: 20px;
}

.demoButton:hover, .signupButton:hover {
    cursor: pointer;
}

.demoButton {
    color: black;
    background-color: transparent;
}

.signupButton {
    background-color: black;
    color: white;
}

.first .left {
    width: 40%;
}

.first .right {
    width: 60%;
}

.section .heading {
    font-size: 38px;
}

.section .subheading {
    font-size: 18px;
}

.mockupDesktop {
    height: auto;
    width: 100%;
}

.icons {
    background-color: #f7f9fa;
}

.iconList {
    display: flex;
    padding: 0px 30px;
    z-index: 30;
}

.icons h2 {
    font-size: 35px;
    z-index: 30;
    /* align-self: flex-start;
    margin-left: 80px; */
}

.iconImage {
    height: auto;
    width: 100%;
}

.iconSection, .last {
    display: flex;
    flex-direction: column;
    align-items: center;
    margin: 50px;
}

.iconHeading {
    align-self: flex-start;
    margin: 30px 0px 0px 0px;
    font-weight: bold;

}

.iconSubheading {
    width: 100%;
    margin: 10px 15px;
    display: block;
}

.security {
    background-color: #F5F1EA;
}

.security .left {
    width: 60%;
    display: flex;
    flex-direction: row;
}

.security .right {
    width: 40%;
}

.securityMockup {
    width: 95%;
    height: auto;
    align-self: flex-start;
}

.last {
    text-align: center;
    margin-top: 10px;
    padding-bottom: 10px;
}

.last .heading {
    font-size: 28px;
}

.footer {
    padding: 5px 30px;
    background-color: #494f5c;
    display: flex;
    justify-content: space-between;
    color: white;
}

.footerDetails {
    display: flex;
    align-items: center;
}

.footer p {
    margin-right: 20px;
}

.footer i {
    font-size: 25px;
    color: white;
    margin: 0px 8px;
}

.footer img {
    height: 70px;
    width: auto;
}

.shape1 {
    position: absolute;
    top: 170vh;
    left: -150px;
    z-index: 0;
    height: 600px;
    width: 550px;
    transform: rotate(-10deg)
}

.shape2 {
    position: absolute;
    z-index: 0;
    top: 220vh;
    height: 700px;
    right: -130px;
}
