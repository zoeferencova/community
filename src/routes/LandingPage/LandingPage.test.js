import React from "react";
import { render } from "enzyme";
import LandingPage from "./LandingPage";
import { BrowserRouter } from "react-router-dom";

describe("LandingPage", () => {
    it("renders LandingPage component given mock context data", () => {
        const wrapper = render(
            <BrowserRouter>
                <LandingPage />
            </BrowserRouter>
        );
        
        expect(wrapper).toMatchSnapshot();
    });
});