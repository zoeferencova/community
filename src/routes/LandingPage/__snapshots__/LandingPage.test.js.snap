// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage renders LandingPage component given mock context data 1`] = `
<main
  class="main"
>
  <header
    class="header"
  >
    <div
      class="left"
    >
      <h1
        class="heading"
      >
        Coming together in a time of need
      </h1>
      <p
        class="subheading"
      >
        Right now, more than ever, we need to unite our communities and help those in need. CommUnity is a platform that enables the sharing of services and resources to ensure no one is left stranded during the Coronavirus (COVID-19) outbreak.
      </p>
      <div
        class="buttonSection"
      >
        <button
          class="signupButton"
          type="button"
        >
          Start now
        </button>
        <button
          class="demoButton"
          type="button"
        >
          See a demo
        </button>
      </div>
    </div>
    <div
      class="right"
    >
      <img
        alt="grocery delivery"
        class="banner desktop"
        src="banner.jpeg"
      />
    </div>
  </header>
  <section
    class="section first"
  >
    <div
      class="left"
    >
      <h2
        class="heading"
      >
        How does it work?
      </h2>
      <p
        class="subheading"
      >
        Post offers or requests for help and see other active postings in your neighborhood. 
        <br />
        <br />
         Other members of your community can then offer to help or accept your offer, initiating a private chat where you can securely work out the details.
      </p>
    </div>
    <div
      class="right"
    >
      <img
        alt="mockup of application on desktop browser"
        class="mockupDesktop"
        src="mockup1.png"
      />
    </div>
  </section>
  <section
    class="section icons"
  >
    <h2>
      What our helpers can do for you
    </h2>
    <div
      class="iconList"
    >
      <div
        class="iconSection"
      >
        <img
          alt="man handing supplies to older couple"
          class="iconImage"
          src="supplies.jpg"
        />
        <div
          class="iconColumn"
        >
          <h3
            class="iconHeading"
          >
            Pick up supplies
          </h3>
          <p
            class="iconSubheading"
          >
            Running low on groceries, medicine, or household supplies? Our helpers can grab anything you need and drop it off at your doorstep.
          </p>
        </div>
      </div>
      <div
        class="iconSection"
      >
        <img
          alt="woman walking dog wearing a face mask"
          class="iconImage"
          src="dog.jpg"
        />
        <div
          class="iconColumn"
        >
          <h3
            class="iconHeading"
          >
            Walk your dog
          </h3>
          <p
            class="iconSubheading"
          >
            Our helpers can take your furry friend for a safe stroll if you donâ€™t feel comfortable being out and about.
          </p>
        </div>
      </div>
      <div
        class="iconSection"
      >
        <img
          alt="decorative shape"
          class="shape1"
          src="shape1.png"
        />
        <img
          alt="decorative shape"
          class="shape2"
          src="shape2.png"
        />
        <img
          alt="woman talking on the phone looking at laptop"
          class="iconImage"
          src="phone.jpg"
        />
        <div
          class="iconColumn"
        >
          <h3
            class="iconHeading"
          >
            Friendly chat
          </h3>
          <p
            class="iconSubheading"
          >
            We know that quarantining can be an isolating and frightening time. Our helpers can offer a friendly chat through our secure instant messaging system.
          </p>
        </div>
      </div>
    </div>
  </section>
  <section
    class="section security"
  >
    <div
      class="left"
    >
      <img
        alt="mockup of chat functionality and post detail page"
        class="securityMockup"
        src="security-mockups.png"
      />
    </div>
    <div
      class="right"
    >
      <h2
        class="heading"
      >
        Your safety and security are important to us
      </h2>
      <p
        class="subheading"
      >
        We will never share any of your personal information. The other users in your neighborhood will only see your first name and location radius (not your exact location), so the amount of information that you share in the chat is 100% your call.
      </p>
    </div>
  </section>
  <section
    class="section last"
  >
    <h2
      class="heading"
    >
      Ready to make a difference in your community?
    </h2>
    <div
      class="buttonSection"
    >
      <button
        class="signupButton"
        type="button"
      >
        Start now
      </button>
      <button
        class="demoButton"
        type="button"
      >
        See a demo
      </button>
    </div>
  </section>
  <footer
    class="footer"
  >
    <div
      class="footerDetails"
    >
      <p>
        Made by Zoe Ferencova
      </p>
      <a
        href="https://github.com/zoeferencova/"
        target="_blank"
      >
        <i
          class="fab fa-github"
        />
      </a>
      <a
        href="https://www.linkedin.com/in/zoeferencova/"
        target="_blank"
      >
        <i
          class="fab fa-linkedin"
        />
      </a>
    </div>
    <img
      alt="small logo"
      class="securityMockup"
      src="light-logo.png"
    />
  </footer>
</main>
`;
